### PR TITLE
[#12447][fix] Correct prompt logprob target token ID mapping

### PR DESCRIPTION
**Pull request by edenfunf**
## Summary

Fixes a bug where prompt logprob entries include the logprob of the **input** token instead of the **next** (predicted) prompt token when that token falls outside the top-K.

- `context_logits[t]` is the model's prediction distribution for position `t+1`, so the target token should be `prompt_token_ids[t+1]`, not `prompt_token_ids[t]`
- Shifts `prompt_token_ids` by one before computing prompt logprobs so each position maps to the correct target
- Last prompt position (predicting the first generation token) returns only the top-K distribution since there's no next prompt token
- Generation logprobs path is unchanged — the mapping there was already correct

**Before (broken):** prompt `[15191, 525, 498, 30]` → position 0 logprobs include `{15191: logprob}` (the input token)  
**After (fixed):** position 0 logprobs include `{525: logprob}` (the next token the model was predicting)

Fixes #12447

## How to reproduce

```bash
python examples/llm-api/quickstart_advanced.py \
  --prompt_logprobs 3 --prompt "Who are you?" \
  --pp_size 1 --tp_size 1 --model_dir <MODEL_DIR>
```

With the fix, the extra token in each prompt logprob entry now correctly corresponds to the next prompt token rather than the input token.

## Test plan

- [x] Updated `test_sampled_token_always_in_prompt_logprobs` to validate target is the next prompt token
- [x] Updated `test_logprobs_against_logits` to shift context tokens for prompt logprobs validation
- [x] Updated `test_llm_return_logprobs` assertion to allow K or K+1 entries (consistent with generation logprobs)
- [x] Verified fix with standalone unit test using mock logits (all 6 scenarios pass)

### CODE REVIEW

**Comment by edenfunf** [comment]
@zhenhuaw-me Hi, could you please help review this PR when you have time? Thanks!

**Comment by zhenhuaw-me** [comment]
@edenfunf I am sorry for the late response, I was OOO.

@Superjomn could you please help to review? I am not familiar with these code. Thanks!

**Comment by edenfunf** [comment]
@Superjomn Hi! When you have a chance, could you take a look at this PR? Thanks!

**Comment by edenfunf** [comment]
Closing this PR — the issue it targets was already fixed upstream by #12499, which shifts `prompt_token_ids` at the call site in `_compute_pytorch_prompt_logprobs` (`base_worker.py`) instead of inside `compute_logprobs`. Issue #12447 has been closed accordingly.

Thanks @zhenhuaw-me and @Superjomn for your time.
